modules/amazon/index.js

```diff
@@ -24,12 +24,16 @@ var extractAsin = function(url) {
 };
 
 var generateErrorMessage = function(error) {
-  var err = error[0].Error[0];
-  var msg;
-  // AWS.InvalidParameterValue is returned when an prospective ASIN isn't actually a real ASIN, so we squash it
-  if (err.Code[0] != "AWS.InvalidParameterValue")
-    msg = "Error: " + err.Message[0];
-  return msg;
+  try {
+    let err = error[0].Error[0];
+    // AWS.InvalidParameterValue is returned when an prospective ASIN isn't actually a real ASIN, so we squash it
+    if (err.Code[0] == "AWS.InvalidParameterValue") {
+      return false;
```

Comment:
for the love of fj please make this `''` instead

Reply:
Hey, look ma, no type system